etc/taskcluster/decisionlib.py

```diff
@@ -563,6 +564,28 @@ def with_features(self, *names):
         self.features.update({name: True for name in names})
         return self
 
+    def with_curl_script(self, url, file_path):
+        self.curl_scripts_count += 1
+        n = self.curl_scripts_count
+        return self \
+        .with_env(**{
+            "CURL_%s_URL" % n: url,
```

Comment:
Is the `n` for making sure we don't end up caching distinct fetches? Can you leave a comment to that effect?

Reply:
If this method is called more than once for the same task definition, we want distinct environment variables variables since those variables end up in the same dictionary. This method itself doesn’t do caching, though.